nixos/modules/virtualisation/containers.nix

```diff
@@ -463,10 +463,15 @@ in
                 A specification of the desired configuration of this
                 container, as a NixOS module.
               '';
-              type = lib.mkOptionType {
+              type = let
+                confPkgs = if config.pkgs == null then pkgs else config.pkgs;
+              in lib.mkOptionType {
                 name = "Toplevel NixOS config";
-                merge = loc: defs: (import ../../lib/eval-config.nix {
+                merge = loc: defs: (import (confPkgs.path + "/nixos/lib/eval-config.nix") {
                   inherit system;
+                  pkgs = confPkgs;
+                  baseModules = import (confPkgs.path + "/nixos/modules/module-list.nix");
+                  inherit (confPkgs) lib;
```

Comment:
Shouldn't this be determined by the `eval-config.nix` from the custom `nixpkgs`?

Reply:
Ideally yes, but as older versions of nixpkgs didn't do this we need to pass all the arguments to stay compatible with older channels.

Reply:
We could make this change in `eval-config.nix` and reconsider after a few years?